code/modules/mob/living/carbon/damage_procs.dm

```diff
@@ -128,11 +131,18 @@
  * Arguments:
  * * slot - organ slot, like [ORGAN_SLOT_HEART]
  * * amount - damage to be set to
+ * * required_organtype - targets only a specific organ type if set to ORGAN_ORGANIC or ORGAN_ROBOTIC
  */
-/mob/living/carbon/setOrganLoss(slot, amount)
-	var/obj/item/organ/O = getorganslot(slot)
-	if(O && !(status_flags & GODMODE))
-		O.setOrganDamage(amount)
+/mob/living/carbon/setOrganLoss(slot, amount, required_organtype)
+	var/obj/item/organ/affected_organ = getorganslot(slot)
+	if(!affected_organ || (status_flags & GODMODE))
+		return
+	if(required_organtype && (affected_organ.status != required_organtype))
+		return
+	var/diff = amount - affected_organ.damage
+	if(!diff)
```

Comment:
Seems a bit convoluted
```suggestion
	if(affected_organ.damage == amount)
```

Reply:
Fixed by commits fa57315c121720506396d05897f86734f0e7e36a and 7ab4e29dc1f53281de7258ca6a913baf9692425b